pay-api/src/pay_api/services/payment_account.py

```diff
@@ -272,22 +273,55 @@ def _save_account(cls, account_request: Dict[str, any], payment_account: Payment
         payment_account.bcol_account = account_request.get('bcolAccountNumber', None)
         payment_account.bcol_user_id = account_request.get('bcolUserId', None)
 
-        billable = get_str_by_path(account_request, 'paymentInfo/billable').lower() == 'true'
+        payment_info = account_request.get('paymentInfo')
+        billable = payment_info.get('billable', True)
         payment_account.billable = billable
 
-        # Create account in CFS based on payment method
-        # If payment method is PAD, OB, EFT, WIRE create a cfs account
-        # For government accounts (billable false) create a cfs account for EJV transactions
+        # Steps to decide on creating CFS Account or updating CFS bank account.
+        # Updating CFS account apart from bank details not in scope now.
+        # Create CFS Account IF:
+        # 1. New payment account
+        # 2. Existing payment account:
+        # -  If the account was on DIRECT_PAY and switching to Online Banking, and active CFS account is not present.
+        # -  If the account was on DRAWDOWN and switching to PAD, and active CFS account is not present
+        cfs_account: CfsAccountModel = CfsAccountModel.find_active_by_account_id(payment_account.id) \
+            if payment_account.id else None
         pay_system = PaymentSystemFactory.create_from_payment_method(payment_method=payment_method)
-        cfs_account_details = pay_system.create_account(name=payment_account.auth_account_name,
-                                                        contact_info=account_request.get('contactInfo'),
-                                                        payment_info=account_request.get('paymentInfo'))
-
-        # Create a new cfs account model if there is no existing one
-        # TODO updating the CFS account in CFS needs to be done during PAD and Online Banking tickets
-        cfs_account = CfsAccountModel.find_active_by_account_id(payment_account.id)
-        if not cfs_account:
-            cfs_account = CfsAccountModel()
+        if pay_system.get_payment_system_code() == PaymentSystem.PAYBC.value:
```

Comment:
does the online banking scenario gets covered by this?

Reply:
Yes. There is test cases for online banking as well.